lib/elixir/lib/task/supervised.ex

```diff
@@ -211,26 +215,43 @@ defmodule Task.Supervised do
           case waiting do
             %{^position => {_, {:ok, _} = ok}} -> Map.put(waiting, position, {nil, ok})
             %{^position => {_, :running}} -> Map.put(waiting, position, {nil, {:exit, reason}})
+            %{^position => {_, :timed_out}} -> Map.put(waiting, position, {nil, {:exit, :killed}})
```

Comment:
@fishcakez okay, got it. So one option would be to `Process.exit(task, :kill)` but emit `{:exit, :killed_for_timeout}`, and another option would be to `Process.exit(task, :kill)` and emit `{:exit, reason_task_died}`, did I get it correctly? If so, which one do you think is better?